### DISCUSSION TITLE
How to store an array of json objects in a data template?

### DISCUSSION THREAD

**Discussion by binduHR** [pactumjs/pactum#363] [Q&A] [answered]
I am constantly facing error
<img width="694" alt="image" src="https://github.com/pactumjs/pactum/assets/101322585/e1a53a0b-c472-4406-a255-f68bd1605c68">

My data template structure is 
```js
stash.addDataTemplate({
    connectionSecrets: [
      {
        "id": 589,
        "createdAt": "xxx",
        "updatedAt":"xxx"
      },
     {
       "id": 590,
       "createdAt": "xxx",
       "updatedAt":"xxx"
     }
   ]
});
```

I am using it as 
```js
 .expectJsonLike({'@DATA:TEMPLATE@': 'connectionSecrets'});
 ```
 And the api response body looks like 
```json
  [
    {
      "id": 589,
      "createdAt": "xxx",
      "updatedAt":"xxx"
    },
    {
      "id": 590,
      "createdAt": "xxx",
      "updatedAt":"xxx"
    }
]
 ```

 How can this be resolved?
 
 And while digging further i found this 
 
![image](https://github.com/pactumjs/pactum/assets/101322585/05f5cef4-ccc7-4201-9212-d9881c2a52c6)

**Comment by ASaiAnudeep** [comment, marked as answer]
Hello @binduHR , have you loaded the data templates into the context of your test execution?

For example, if the data templates are declared in a different file, we need to load them before the start of the test execution.

```js
// template.js

stash.addDataTemplate({
    connectionSecrets: [
      {
        "id": 589,
        "createdAt": "xxx",
        "updatedAt":"xxx"
      },
     {
       "id": 590,
       "createdAt": "xxx",
       "updatedAt":"xxx"
     }
   ]
});
```


```test.js
// load template file
require('template.js')

await spec()
  .expectJsonLike('connectionSecrets')
```

**Reply by binduHR** [reply]
Yeah that's a miss but still facing some problems

So we can have our data templates in the follwoing structure right?

stash.addDataTemplate({
    connectionSecrets: [
      {
        "id": 589,
        "createdAt": "xxx",
        "updatedAt":"xxx"
      },
     {
       "id": 590,
       "createdAt": "xxx",
       "updatedAt":"xxx"
     }
   ],
   eventTemplates : [{ },{ } ,{ } ]
   
   
<img width="865" alt="image" src="https://github.com/pactumjs/pactum/assets/101322585/ffd52070-ea1c-4db7-a0cd-f0f2ca24c924">



});

**Reply by binduHR** [reply]
Ignore the issue resolved.Thank you @ASaiAnudeep
